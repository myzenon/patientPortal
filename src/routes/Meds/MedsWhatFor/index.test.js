import React from 'react'
import { MedsWhatFor } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test.skip('renders correctly', () => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "y",
                dosage: {}
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        }
    }
    const wrapper = render(
        <MedsWhatFor { ...props } />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test.skip('populate checkbox follows med indications', done => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "y",
                dosage: {}
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        }
    }
    const wrapper = mount(
        <MedsWhatFor { ...props } />
    )

    setTimeout(() => {
        //expect(toJson(wrapper.find('tr').render())).toMatchSnapshot()
        done()
    }, 0)
})

test('clicks next | go to meds-any-other-questions page', done => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "y",
                dosage: {}
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                medication_000080833: {
                    taken: "y",
                    dosage: {},
                    reasonCode: []
                }
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                medication_000080833: {
                    taken: "y",
                    dosage: {},
                    reasonCode: []
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/meds-any-other-questions/000080833')
            done()
        }
    }
    const wrapper = mount(
        <MedsWhatFor { ...props } />
    )
    wrapper.find('button').simulate('click')
})


test('clicks next and there is smart question for this medication | go to smart-questions-medication', done => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000293159: {
                taken: "y",
                dosage: {}
            }
        },
        match: {
            params: {
                medId: '000293159'
            }
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                medication_000293159: {
                    taken: "y",
                    dosage: {},
                    reasonCode: []
                }
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                medication_000293159: {
                    taken: "y",
                    dosage: {},
                    reasonCode: []
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/smart-questions/medication_000293159/1')
            done()
        }
    }
    const wrapper = mount(
        <MedsWhatFor { ...props } />
    )
    wrapper.find('button').simulate('click')
})
