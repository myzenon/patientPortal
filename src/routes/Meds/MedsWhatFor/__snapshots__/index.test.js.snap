// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <span
      class="heading1"
    >
      EFFEXOR XR 75MG
    </span>
    <p />
    <div>
      <div
        class="imageWrapper"
      >
        <img
          alt=""
          class="contentImg"
          src="https://rximage.nlm.nih.gov/image/images/gallery/original/00008-0833-21_RXNAVIMAGE10_3015981C.jpg"
        />
        <img
          alt="Zoom-In"
          class="imageMagnifyZoomIn"
          src="magnify-in.png"
        />
      </div>
    </div>
    <p />
    <span
      class="questions"
    >
      What did the doctor tell you this medication was for? (Check all that apply)
    </span>
    <p />
    <div
      class="indicationContainer"
    >
      <div
        class="row"
      >
        <div
          class="checkboxWrapper col-1 col-md-1 offset-md-1"
        >
          <label
            class="checkboxContainer"
          >
            <input
              name="Major Depressive Disorder"
              type="checkbox"
            />
            <span
              class="checkbox"
            />
          </label>
        </div>
        <div
          class="formLabel col-9 col-md-9"
        >
          <span
            class="indicationConditionName"
          >
            Major Depressive Disorder
          </span>
          <p />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="checkboxWrapper col-1 col-md-1 offset-md-1"
        >
          <label
            class="checkboxContainer"
          >
            <input
              name="Generalized Anxiety Disorder"
              type="checkbox"
            />
            <span
              class="checkbox"
            />
          </label>
        </div>
        <div
          class="formLabel col-9 col-md-9"
        >
          <span
            class="indicationConditionName"
          >
            Generalized Anxiety Disorder
          </span>
          <p />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="checkboxWrapper col-1 col-md-1 offset-md-1"
        >
          <label
            class="checkboxContainer"
          >
            <input
              name="Vasomotor Symptoms"
              type="checkbox"
            />
            <span
              class="checkbox"
            />
          </label>
        </div>
        <div
          class="formLabel col-9 col-md-9"
        >
          <span
            class="indicationConditionName"
          >
            Vasomotor Symptoms
          </span>
          <p />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="checkboxWrapper col-1 col-md-1 offset-md-1"
        >
          <label
            class="checkboxContainer"
          >
            <input
              name="Social Anxiety Disorder"
              type="checkbox"
            />
            <span
              class="checkbox"
            />
          </label>
        </div>
        <div
          class="formLabel col-9 col-md-9"
        >
          <span
            class="indicationConditionName"
          >
            Social Anxiety Disorder
          </span>
          <p />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="checkboxWrapper col-1 col-md-1 offset-md-1"
        >
          <label
            class="checkboxContainer"
          >
            <input
              name="Neuropathic Pain"
              type="checkbox"
            />
            <span
              class="checkbox"
            />
          </label>
        </div>
        <div
          class="formLabel col-9 col-md-9"
        >
          <span
            class="indicationConditionName"
          >
            Neuropathic Pain
          </span>
          <p />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="checkboxWrapper col-1 col-md-1 offset-md-1"
        >
          <label
            class="checkboxContainer"
          >
            <input
              name="I don't know"
              type="checkbox"
            />
            <span
              class="checkbox"
            />
          </label>
        </div>
        <div
          class="formLabel col-9 col-md-9"
        >
          <span
            class="indicationConditionName"
          >
            I don't know
          </span>
          <p />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="checkboxWrapper col-1 col-md-1 offset-md-1"
        >
          <label
            class="checkboxContainer"
          >
            <input
              name="Other"
              type="checkbox"
            />
            <span
              class="checkbox"
            />
          </label>
        </div>
        <div
          class="formLabel col-9 col-md-9"
        >
          <input
            class="inputText"
            name="inputOther"
            placeholder="Enter Other here"
            type="text"
            value=""
          />
          <p />
        </div>
      </div>
    </div>
    <br />
    <p>
      <button
        class="button btn btn-secondary"
        id="MedsWhatForButton"
        type="button"
      >
        <b>
           NEXT
        </b>
      </button>
    </p>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepCompleted"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="stepActive"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumberActive"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
